### ISSUE TITLE
Stored Procedure Properties Source tab randomly adds new line characters

### ISSUE THREAD

**Issue by jmccay-work** [dbeaver/dbeaver#10578]
I first wanted to say, great program!  I love it.  You have made working with Informix a lot better!  With that said, an acquaintance and I noticed that there is new line characters added at odd positions within the middles of words.  There shouldn't be any new line characters.

Below, I have pasted a small snippet showing the problem.

I am using version the community Version 7.2.4.202011011603 released on 11/2/2020, but I have seen similar problems with this in the past.

The data source is a JDBC connection to an Informix Server using the jar files the software downloads on Windows systems.

```
 newactnum like cadactivities.activitynum,
  newdispassign like cadactivities.dispassign,
  ne
wfromassign like cadactivities.fromassign,
```

**Comment by jmccay-work** [comment]
`
```
  newactnum like cadactivities.activitynum,
  newdispassign like cadactivities.dispassign,
  ne 
wfromassign like cadactivities.fromassign,

````

**Comment by LonwoLonwo** [comment]
Hello @jmccay-work 
Could you give an example of a DDL procedure that is displayed incorrectly? Entirely. For testing.

Currently, line breaks in a procedure source are used to separate multiple procedures with the same name but different parameters.
![2020-12-02 10_42_49-DBeaver 7 3 0 - ifx_set_erstate](https://user-images.githubusercontent.com/45152336/100843451-6d131e00-348b-11eb-8d04-475c9f6c56ff.png)

**Comment by jmccay-work** [comment]
Sorry it has taken me so long to respond, but I got busy yesterday.  I copied the a stored procedure 2 times.  One of them I used dbschema to produce correctly, and the other I copied the text from dbeaver.  Please ignore the bad structural formating.  Some of my colleagues don't care about making it look good.  A lot of this is legacy stuff from long before me.

The good one:
```
drop procedure if exists acttolog;
create procedure "uqed".acttolog(
  newlogaction char(1),
  newsernum integer,
  newagency like cadactivities.agencyname,
  newdeploy like cadactivities.deploymentnum,
  newactnum like cadactivities.activitynum,
  newdispassign like cadactivities.dispassign,
  newfromassign like cadactivities.fromassign,
  newcoverassign like cadactivities.coverassign,
  newacttype like cadactivities.activitytype,
  newdtstarted like cadactivities.dtstarted,
  newdtonloc like cadactivities.dtonloc,
  oldstatus char(15),
  newstatus like cadactivities.status,
  newdtstatus like cadactivities.dtstatus,
  oldavail char(1),
  newavail like cadactivities.availinstatus,
  oldstatusloc char(15),
  newstatusloc like cadactivities.statusloc,
  newincnum like cadactivities.incnum,
  newdreqnum like cadactivities.dreqnum,
  newcuractnum like cadactivities.curactnum,
  newlocname like cadactivities.locname,
  newdispid like cadactivities.dispid
)

  -- If this is an insert
  if newlogaction == "I" and newincnum is not null
      and newdreqnum is not null then

    -- Enforce the rule that an activity cannot be added for a closed request
    -- If there is any incident, and if the request is open
    if (select archivestatus from caddispreq where incnum = newincnum
           and dreqnum = newdreqnum) <> "O" then

      raise exception -9961, 0,
         "Incident has been recently updated-- please check and re-dispatch.";

    end if

  end if
 
  -- For inserts or updates
  if newlogaction in ("I", "U") and newincnum is not null
       and newdreqnum is not null then

    -- Make sure there are no other activities already associated with the dispatch request
    if exists (select * from cadactivities where incnum = newincnum
                   and dreqnum = newdreqnum
                   and sernum != newsernum) then

      raise exception -9958, 0, "Cannot dispatch two activities to one dispatch request-- please check and re-dispatch.";

    end if

  end if

  -- Log this change to cadlog
  insert into cadlog (logdt, logtable, logkey, logaction,
                      agencyname, deploymentnum, activitynum,
                      dispassign, vehicleid, coverassign,
                      inctype, dtreceived, dtscheduled,
                      status, dtstatus, availinstatus,
                      statusloc, incnum, dreqnum, shiftid, locname, userid)
  values (current year to second, "cadactivities", newsernum, newlogaction,
                      newagency, newdeploy, newactnum,
                      newdispassign, newfromassign, newcoverassign,
                      newacttype, newdtstarted, newdtonloc,
                      newstatus, newdtstatus, newavail,
                      newstatusloc, newincnum, newdreqnum,
                      newcuractnum, newlocname, newdispid);

  -- If this is an insert, or an update with new status information
  if (newlogaction == "I") or (newlogaction = "U"
        and (newstatus <> oldstatus or newavail <> oldavail
              or newstatusloc <> oldstatusloc)) then

    -- Also log it to the agency status log
    insert into cadagencystatuses (agencyname, deploymentnum, activitynum,
                        status, dtstatus,
                        availinstatus, statusloc, dispid)
    values (newagency, newdeploy, newactnum,
                       newstatus, newdtstatus,
                       newavail, newstatusloc, user);
  end if

end procedure;
```
The problem source copied from dbeaver:
```
create procedure acttolog(
  newlogaction char(1),
  newsernum integer,
  newagency like cadactivities.agencyname,
  newdeploy like cadactivities.deploymentnum,
  newactnum like cadactivities.activitynum,
  newdispassign like cadactivities.dispassign,
  ne
wfromassign like cadactivities.fromassign,
  newcoverassign like cadactivities.coverassign,
  newacttype like cadactivities.activitytype,
  newdtstarted like cadactivities.dtstarted,
  newdtonloc like cadactivities.dtonloc,
  oldstatus char(15),
  newstatu
s like cadactivities.status,
  newdtstatus like cadactivities.dtstatus,
  oldavail char(1),
  newavail like cadactivities.availinstatus,
  oldstatusloc char(15),
  newstatusloc like cadactivities.statusloc,
  newincnum like cadactivities.incnum,
  newdreqn
um like cadactivities.dreqnum,
  newcuractnum like cadactivities.curactnum,
  newlocname like cadactivities.locname,
  newdispid like cadactivities.dispid
)

  -- If this is an insert
  if newlogaction == "I" and newincnum is not null
      and newdreqnum 
is not null then

    -- Enforce the rule that an activity cannot be added for a closed request
    -- If there is any incident, and if the request is open
    if (select archivestatus from caddispreq where incnum = newincnum
           and dreqnum = newdr
eqnum) <> "O" then

      raise exception -9961, 0,
         "Incident has been recently updated-- please check and re-dispatch.";

    end if

  end if
 
  -- For inserts or updates
  if newlogaction in ("I", "U") and newincnum is not null
       and newd
reqnum is not null then

    -- Make sure there are no other activities already associated with the dispatch request
    if exists (select * from cadactivities where incnum = newincnum
                   and dreqnum = newdreqnum
                   and sern
um != newsernum) then

      raise exception -9958, 0, "Cannot dispatch two activities to one dispatch request-- please check and re-dispatch.";

    end if

  end if

  -- Log this change to cadlog
  insert into cadlog (logdt, logtable, logkey, logaction,

                      agencyname, deploymentnum, activitynum,
                      dispassign, vehicleid, coverassign,
                      inctype, dtreceived, dtscheduled,
                      status, dtstatus, availinstatus,
                      st
atusloc, incnum, dreqnum, shiftid, locname, userid)
  values (current year to second, "cadactivities", newsernum, newlogaction,
                      newagency, newdeploy, newactnum,
                      newdispassign, newfromassign, newcoverassign,
     
                 newacttype, newdtstarted, newdtonloc,
                      newstatus, newdtstatus, newavail,
                      newstatusloc, newincnum, newdreqnum,
                      newcuractnum, newlocname, newdispid);

  -- If this is an insert
, or an update with new status information
  if (newlogaction == "I") or (newlogaction = "U"
        and (newstatus <> oldstatus or newavail <> oldavail
              or newstatusloc <> oldstatusloc)) then

    -- Also log it to the agency status log
    i
nsert into cadagencystatuses (agencyname, deploymentnum, activitynum,
                        status, dtstatus,
                        availinstatus, statusloc, dispid)
    values (newagency, newdeploy, newactnum,
                       newstatus, newdtst
atus,
                       newavail, newstatusloc, user);
  end if

end procedure;                                                                                                                                                                            
```

Does this help?  Thank you for the quick response.

**Comment by LonwoLonwo** [comment]
@jmccay-work thanks, it helps.
Column data from sysprocbody is CHAR(256), so long text of procedures does not fit there.
I'll think about what I can do about it.

**Comment by kseniiaguzeeva** [comment]
Verified (procedures were tested with 256 signs and more).

**Comment by LonwoLonwo** [comment]
Fixed in 7.3.3
Will be available [here](https://dbeaver.io/) at 2021-01-18

**Comment by jmccay-work** [comment]
Thanks, I'll check it out.

Sincerely,

Joseph L. McCay
